### PR TITLE
Add and use CMake modules to generate SPIR-V kernels

### LINKED ISSUE #213: CMake equiavlent to Bazel iree_glsl_vulkan/spirv_kernel_cc_library
`iree_glsl_vulkan` (defined [here](https://github.com/google/iree/blob/master/iree/build_defs.bzl#L111-L112)) is aliasing `glsl_vulkan` and finally calls `_glslang` (both defined [here](https://github.com/google/iree/blob/3e572044d1e2014189aa11fd7e7d9f419ab64857/build_tools/third_party/glslang/build_defs.bzl#L17-L42)). In the current CMake implementation, the `glslangValidator` command is embedded into `add_custom_command`. Within Bazel `iree_glsl_vulkan` is called by `spirv_kernel_cc_library` (see [here](https://github.com/google/iree/blob/3e572044d1e2014189aa11fd7e7d9f419ab64857/iree/compiler/Translation/SPIRV/Kernels/spirv_utils.bzl#L20-L46)) to generate the SPIR-V kernels. Instead of using directly defining `add_custom_command` every time, a CMake wrapper closer to Bazel is preferred.

The comment by @benvanik  in `iree/compiler/Translation/SPIRV/Kernels/CMakeLists.txt` mentions to implement this as `iree_glsl_cc_library` for CMake. This would mean we have a slightly different naming between CMake and Bazel. Either we can go close with the Bazel flow or we can handle this all in `iree_glsl_cc_library`. Any thought on this?

Because `generate_cc_embed_data` is called, this depends on #211.

### PR DESCRIPTION
**Pull request by marbre**
Closes #213

### CODE REVIEW

**Comment by GMNGeoffrey** [comment]
Ah actually Scott's OOO today. I'll take a look

File: build_tools/cmake/iree_glslang.cmake [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+# iree_glslang()
+#
+# CMake function to imitate Bazel's _glslang rule.
+#
+# Parameters:
+# NAME: Name of spv file to create, whichout file name extension.
+# SRC: Source file.
+# MODE: Defines the typ of input, either "hlsl" or "glsl".
+# TARGET: Target to create the SPIR-V binary for, either "vulkan" or "opengl".
```
Comment (GMNGeoffrey) [inline build_tools/cmake/iree_glslang.cmake]:
```suggestion
# TARGET: Target to create the SPIR-V binary for: either "vulkan" or "opengl".
```

File: build_tools/cmake/iree_hlsl_vulkan.cmake [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+# limitations under the License.
+
+include(CMakeParseArguments)
+
+# iree_hlsl_vulkan()
+#
+# CMake function to imitate Bazel's iree_hlsl_vulkan rule and hlsl_vulkan rule
+#
+# Parameters:
+# NAME: Name of spv file to create, whichout file name extension.
```
Comment (GMNGeoffrey) [inline build_tools/cmake/iree_hlsl_vulkan.cmake]: Nit: typo and grammar suggestion

```suggestion
# NAME: Name of spv file to create (without the file name extension).
```

File: build_tools/cmake/iree_glsl_vulkan.cmake [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Copyright 2019 Google LLC
```
[the remark this answers is not in the document]
Comment (marbre) [reply build_tools/cmake/iree_glsl_vulkan.cmake]: Well yes, I started in 2019 with that one ;-D Fixed it.

[the remark this answers is not in the document]
Comment (marbre) [reply build_tools/cmake/iree_glsl_vulkan.cmake]: BTW, does this imply to adjust the copyright year to 2019-2020, if old files are changed?

File: build_tools/cmake/iree_glslang.cmake [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+# limitations under the License.
+
+include(CMakeParseArguments)
+
+# iree_glslang()
+#
+# CMake function to imitate Bazel's _glslang rule.
+#
+# Parameters:
+# NAME: Name of spv file to create, whichout file name extension.
```
[the remark this answers is not in the document]
Comment (marbre) [reply build_tools/cmake/iree_glslang.cmake]: Thanks for catching this. Done.

**Pull request closed without merging**